src/hotspot/share/opto/vectornode.cpp

```diff
@@ -752,7 +752,16 @@ Node* StoreVectorMaskedNode::Ideal(PhaseGVN* phase, bool can_reshape) {
       return phase->transform(new StoreVectorNode(ctr, mem, adr, adr_type(), val));
     }
   }
-  return NULL;
+  return StoreNode::Ideal(phase, can_reshape);
+}
+
+const TypeVect* StoreVectorMaskedNode::vect_type() const {
+  Node* value = in(MemNode::ValueIn);
+  if (value->bottom_type()->isa_vect() != NULL) {
+    return value->bottom_type()->is_vect();
+  }
+  // The value input is actually a BinaryNode, which has two inputs value and mask.
```

Comment:
BinaryNodes are created just before selection. In other cases a vector binary operation should be store's input value and its bottom_type() must be a valid vector type. Did you see any failure/issue for which this special handling for Binarynode is added?

Reply:
Yes, this issue is caused due to the matcher puts the third and forth inputs into a BinaryNode before matching. See: https://github.com/openjdk/panama-vector/blob/vectorIntrinsics/src/hotspot/share/opto/matcher.cpp#L2348

The issue will happen when `vect_type()` is called after then (i.g. in the ad file).

Reply:
I guess the latest code doesn't have this issue, due to https://github.com/openjdk/jdk/pull/2867. I will update the codes and have a test. Thanks!